src/components/settings/PersonalizedPushSettings.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useContext, useEffect, useMemo, useState } from 'react';
+import { useQuery } from 'react-apollo';
+import { FlatList, StyleSheet } from 'react-native';
+
+import { SettingsContext } from '../../SettingsProvider';
+import { normalize, texts } from '../../config';
+import { getInAppPermission } from '../../pushNotifications';
+import { getQuery, QUERY_TYPES } from '../../queries';
+import { onActivatePushNotifications, onDeactivatePushNotifications } from '../../screens';
+import { LoadingSpinner } from '../LoadingSpinner';
+import { SafeAreaViewFlex } from '../SafeAreaViewFlex';
+import { SettingsToggle } from '../SettingsToggle';
+
+type Category = { id: string; iconName?: string | null; name: string };
+
+const keyExtractor = (item, index) => `index${index}-id${item.id}`;
+
+export const PersonalizedPushSettings = () => {
+  const { globalSettings } = useContext(SettingsContext);
+  const { settings = {} } = globalSettings;
+  const { personalizedPush = {}, settingsScreenListItemTitles = {} } = settings;
+  const queryVariables = personalizedPush.variables || {
+    tagList: ['news_item']
+  };
+
+  const [permission, setPermission] = useState<boolean>(false);
+
+  const { data, loading } = useQuery(getQuery(QUERY_TYPES.CATEGORIES_FILTER), {
+    variables: queryVariables,
+    skip: !personalizedPush?.categories
+  });
+
+  useEffect(() => {
+    (async () => {
+      const inAppPermission = await getInAppPermission();
+      setPermission(!!inAppPermission);
+    })();
+  }, []);
+
+  const onActivate = (revert: () => void) => {
+    setPermission(true);
+    onActivatePushNotifications(() => {
+      setPermission(false);
+      revert?.();
+    });
+  };
+
+  const onDeactivate = (revert: () => void) => {
+    setPermission(false);
+    onDeactivatePushNotifications(() => {
+      setPermission(true);
+      revert?.();
+    });
+  };
+
+  const listItems = useMemo(() => {
+    const categories: Category[] = data?.categories ?? [];
+
+    const initialListItem = {
+      onActivate,
+      onDeactivate,
+      title:
+        settingsScreenListItemTitles.pushNotifications || texts.settingsTitles.pushNotifications,
+      topDivider: false,
+      value: permission
+    };
+
+    if (!categories?.length) return [initialListItem];
+
+    const parsedListItems = categories.map((category) => ({
+      categoryId: category.id,
+      iconName: category.iconName,
+      id: category.id,
+      isDisabled: !permission,
+      // TODO: the necessary functions to activate and deactivate the push by category will be updated
```

Comment:
The connection to the server is added with the commit here. -> fc157e4d72b24d1c5a90ff9b2d334c2cc5b1e1cf